src/edu/csus/ecs/pc2/core/model/RunFiles.java

```diff
@@ -178,16 +182,88 @@ public String toString() {
                 }
 
             }
-            
-        } 
+
+        }
         //close OtherFiles list
-        retStr += "]";  
-        
+        retStr += "]";
+
         //close retStr object
         retStr += "]";
-        
+
         return retStr ;
-        
+
     }
 
+    /**
+     * createCatalogJSON - Generate detailed JSON information about the files submitted and write it to the
+     * supplied BufferedWriter.  Similar to toString above, but supplies more details.
+     *
+     * @param catlog Where to send the detailed JSON about the submission files
+     */
+    public void createCatalogJSON(BufferedWriter catlog, Log log) {
+        try {
+            boolean needComma = false;
+            catlog.write("[");
+            if(mainFile != null) {
+                catlog.newLine();
+                catlog.write(" {\"mainfile\":");
+                catlog.write(serializedFileToJSON(mainFile));
+                catlog.write(" }");
+                needComma = true;
+            }
+            if(otherFiles != null && otherFiles.length > 0) {
+                boolean needComma2 = false;
```

Comment:
I don't see any reference/use of `needComma2` in the code.  Am I missing something?

Reply:
This was a bug in my loop.  The loop should have used `needcomma2 `not `needcomma`.   Basically, if there's more than one "other file", then we have to put commas between those.  The other files are in a separate json object, so they have their own separators (`needComma2`).

I fixed the loop.